File: CivitasJava/civitas/Sorpresa_IrCarcel.java
package civitas;

import java.util.ArrayList;

class Sorpresa_IrCarcel extends Sorpresa
{
    private Tablero tablero;

    Sorpresa_IrCarcel(Tablero tablero)
    {
        super("Vaya a la cárcel.");
        this.tablero=tablero;
    }

    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if(super.jugadorCorrecto(actual, todos))
        {
            informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }

    @Override
    public String toString()
    {
        return "Sorpresa que envía a la cárcel.";
    }
}

File: CivitasRuby/Sorpresa_IrCarcel.rb
module Civitas
    class Sorpresa_IrCarcel < Sorpresa
        public_class_method :new
        
        def initialize(tablero)
            super('Vaya a la carcel.')
            @tablero=tablero
        end

        def aplicarAJugador(actual, todos)

            if jugadorCorrecto(actual,todos)
                informe(actual, todos)
                todos[actual].encarcelar(@tablero.numCasillaCarcel)
            end
        end

        def toString
            return 'Sorpresa que envia a la carcel.'
        end

    end
end